xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -34,40 +34,37 @@ async def _wait_for_final_transaction_outcome(
     await asyncio.sleep(_LEDGER_CLOSE_TIME)
     # new persisted transaction
 
+    latest_ledger_sequence = await get_latest_validated_ledger_sequence(client)
+
+    if last_ledger_sequence < latest_ledger_sequence:
```

Comment:
This needs to be flipped

Reply:
This flip is actually correct (Although the names are terrible, I'll try to find better names)

When you create a transaction, you set `LastLedgerSequence` to current ledger + 20 or something. 

So in the first iteration:
`LastLedgerSequence: +20` | `latest_ledger_sequence: 0`
Which means that we skip over the error and try again later, as expected. 

20 | 1 
20 | 2
...
20 | 21 -> `LastLedgerSequence` < `latest_ledger_sequence` -> Error, waited too long.

(And this is a nit, but it's not `<=` because it's supposed to be ok to validate if the ledger is on, or before that index, so we shouldn't error if we're on an "equal to" ledger index - https://xrpl.org/reliable-transaction-submission.html#:~:text=LastLedgerSequence%20is%20an%20optional%20parameter,than%20the%20transaction's%20LastLedgerSequence%20parameter.)

Reply:
The flip happened because I had to flip the if-statement to do the error before the recursion, since the error is the base case now (and otherwise it just goes infinite, never checking the ledger index)

Reply:
I'd recommend flipping it entirely then to make the logic easier to parse - 
```py
if latest_ledger_sequence > last_ledger_sequence:
```
Since `latest_ledger_sequence` is changing and `last_ledger_sequence` is not

Reply:
I think the name `current_ledger_sequence` is better than `latest` because it's too close to `last`. I'll go with that unless that seems confusing for some reason

Reply:
(And my nit was wrong - I thought it was the index of the current ledger, but it was the last validated ledger, which means that if we already see "LastLedgerSequence" there it's too late to be included)